spec/controllers/links_controller_spec.rb

```diff
@@ -41,47 +41,149 @@
     end
   end
 
-  describe '#update' do
-    it 'updates link and redirects to HTTP_REFERER' do
-      link = Link.create(given_url: Faker::Internet.url,
-                         slug: Faker::Internet.slug)
-      post :update, id: link.id,
-                    slug: 'some',
-                    given_url: link.given_url,
-                    active: true
-      expect(response.status).to eq 302
-      expect(flash[:notice]).to be_present
+  describe 'GET #show' do
+    let(:user) { create(:user) }
+    let(:link) { create(:link) }
+
+    context 'when user is signed in' do
+      before do
+        user.links << link
+        session[:id] = user.id
+        get :show, link_id: link.id
+      end
+
+      it 'displays the dashboard' do
+        expect(response).to render_template 'show'
+      end
+
+      it 'has an array containing my links' do
+        expect(assigns(:links)).to_not be_nil
+        expect(assigns(:link)).to_not be_nil
+      end
     end
-  end
 
-  describe '#activate' do
-    before do
-      @link = FactoryGirl.create(:link, active: false)
+    context 'when user is not signed in' do
+      before { get :show }
+      it 'redirects to sign in page' do
+        expect(response).to redirect_to sign_in_path
+      end
     end
-    it 'sets the link status to active' do
-      post :toggle_activate, id: @link, active: true
-      expect(response.status).to eq 302
-      expect(flash[:notice]).to be_present
+  end
+
+  describe 'PATCH #update' do
+    let(:link) { create(:link) }
+    let(:user) { create(:user) }
+
+    context 'when user has a valid session' do
+      before do
+        request.env['HTTP_REFERER'] = dashboard_path
+        session[:id] = user.id
+        patch :update, id: link.id, link: { slug: 'some' }
+      end
+
+      it 'updates link and goes back to dashboard' do
```

Comment:
**Critical - Test Documentation**
The description states `updates link and go back to dashboard`. However, I can only see the expectation for redirecting to the dashboard.

As much as possible, your descriptions should be in line with what's actually been tested.

Kindly add the other expectation i.e. checking that the link was actually updated.